src/init.cpp

```diff
@@ -1394,7 +1375,9 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
                 }
 
                 // Check for changed -insightexplorer state
-                if (fInsightExplorer != GetBoolArg("-insightexplorer", false)) {
+                bool fInsightExplorerPreviouslySet = false;
+                pblocktree->ReadFlag("insightexplorer", fInsightExplorerPreviouslySet);
```

Comment:
I don't think the branch currently compiles, because `LoadBlockIndexDB()` in `main.cpp` still contains:
```
    // insightexplorer
    // Check whether block explorer features are enabled
    pblocktree->ReadFlag("insightexplorer", fInsightExplorer);
    LogPrintf("%s: insight explorer %s\n", __func__, fInsightExplorer ? "enabled" : "disabled");
    fAddressIndex = fInsightExplorer;
    fSpentIndex = fInsightExplorer;
    fTimestampIndex = fInsightExplorer;
```
but `fInsightExplorer` no longer exists.

I think the solution is just to use a local in `LoadBlockIndexDB()`, same as here. `pblocktree` should be valid here, because we were previously relying on `fInsightExplorer` being set here by `LoadBlockIndexDB()`.

Reply:
Can you clarify please ? As the branch is actually building it is not clear what is the suggested change. Thanks.

Reply:
I think i got it. changed `fInsightExplorer` to a local at https://github.com/zcash/zcash/pull/4293/commits/aa24f921e8ef5737d2a516c681953fa5592af664 .

Also rebased with the last `master`.